src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,32 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private final Random random = new Random();
+    private final ColorSupplier randomColor = new ColorSupplier();
+    private final int randomFigureType = random.nextInt(5);
+    private final int randomSide = random.nextInt() * 16;
+
+    public Figure getRandomFigure() {
+        switch (randomFigureType) {
+            case 0:
+                return new Square(randomSide, randomColor.getRandomColor());
+            case 1:
+                return new Rectangle(randomSide, randomSide * 2, randomColor.getRandomColor());

```

Comment:
I would not say, that multiplication by 2 is a good random strategy. I recommend to generate each time unique value per each side.